### ISSUE TITLE
Support TAP output

### ISSUE THREAD

**Issue by garethr** [instrumenta/conftest#59]
With the work started in #55 we are soon to be able to support multiple output formats. Some discussion in #26 and #42. 

Opening this issue to discuss TAP output support. I think the best we can do is:

```tap
1..3
not ok 1 - deployment.yaml - Containers must not run as root
not ok 2 - deployment.yaml - Deployments are not allowed
# Warnings
not ok 3 - deployment.yaml - Services not allowed
```

1. We can currently only show failing rules, so no `ok` responses at present
2. I think we should include Warnings in the output. The `#` is a diagnostic information, it's not machine readable be is at least human readable. The alternative would be to not include the warnings which I feel is worse
3. Each line includes the file and the message, similar to #58 

The TAP specification can be found at http://testanything.org/tap-specification.html

TAPs advantage is its simplicity, so a number of tools can consume it and display the results in whatever way they choose.

**Comment by garethr** [comment]
PR in #68

**Comment by garethr** [comment]
@brendanjryan would you mind taking a look at this PR? This adds another output format to the ones you'd already implemented.

**Comment by garethr** [comment]
Merged #68 with this feature.
